runbooks/how-to-load-cloudwatch-historical-data.md

```diff
@@ -0,0 +1,223 @@
+      overlap will be accounted for when populating the `staging` table.
+   2. Navigate to Cloudwatch in the AWS console.
+   3. Select `Log Groups`
+   4. Select `/bfd/prod/bfd-server/access.json`
+   5. Select `Actions -> Export Data to Amazon S3`
+   6. Choose the time period 
+   7. Select Account: *This Account*
+   8. S3 Bucket Name: `bfd-insights-bfd-app-logs`
+   9. S3 Bucket Prefix: `export/prod/YYYY_(MM-MM)` (For example, export/prod/2022_(01-06) for the data that spans from 
+      2022-01-01 00:00:00 through 2022-07-01 00:00:00)
+   10. Run the job. Production exports can take up to an hour per month depending on the activity level that month.
+   11. When the job completes, remove the `aws-logs-write-test` subfolder that AWS creates for internal testing.
+
+2. Create/Update the Glue `export` table.
+   1. From the AWS Glue console, select Crawlers
+   2. Select the `bfd_cw_export` crawler (this crawler is in terraform and should already exist)
+   3. Select `Run`
+   4. Verify that the `bfd_cw_export.prod` table has the expected data by inspecting the results of this query that
+      retrieves the number of records per month and ensuring that the newly exported data is represented:
+   
+   ```sql
+   select date_format(from_iso8601_timestamp("timestamp"), '%Y-%m'), count(*)
+   from bfd_cw_export.prod
+   group by 1
+   order by 1
+   ```
+
+3. Determine the column list for the `staging` table.
+   1. The column list for the `staging` table should be identical in names, data types, and ordering to the `target`
+      table after removing the `year` and `month` partition columns. The column definition for the `target` table can be
+      retrieved by navigating to the table in the AWS Glue console, selecting `Actions` and then `View properties` which
+      makes a JSON schema for the table available which includes the ordered list of columns. The column names can be
+      extracted from this file and should resemble the list below. Only include the column names from the `Columns`
+      array -- do not include `PartitionKeys`.
+
+      Sample output:
+      ```text
+      cw_timestamp
+      cw_id
+      timestamp
+      level
+      thread
+      logger
+      message
+      context
+      mdc_bene_id
+      mdc_database_query_bene_by_coverage_batch
+      ... <over 200 additional mdc columns>
+      ```
+   2. The column list for the `staging` table is constructed from the output from step 2 adding the column type (all
+      columns are of string type).
+   ```text
+      cw_timestamp string,
+      cw_id string,
+      timestamp string,
+      level string,
+      thread string,
+      logger string,
+      message string,
+      context string,
+      mdc_bene_id string,
+      mdc_database_query_bene_by_coverage_batch string,
+      ... <all other mdc columns in order>
+   ```
+
+5. Create a non-partitioned Parquet Glue table to serve as the `staging` table.
```

Comment:
Minor: numbering is still messed up, we go from 3 -> 5 (maybe cause you removed something)